fission-web-server/library/Fission/Web/Server/IPFS/Cluster/Pin/Status/Types.hs

```diff
@@ -0,0 +1,30 @@
+module Fission.Web.Server.IPFS.Cluster.Pin.Status.Types (Status (..)) where
+
+import           Fission.Prelude
+
+data Status
+  = FailedWith Text
+  | Unexpected Text
+  | Queued
+  | Pinning
+  | Pinned
+  deriving (Eq, Show)
+
+instance FromJSON Status where
+  parseJSON = withObject "Cluster.Pin.Status" \obj -> do
+    status   <- obj .: "status"
+    errorTxt <- obj .: "error"
+
+    case status :: Text of
+      "pinned"       -> return Pinned
+
+      "pinning"      -> return Pinning
+
+      "pin_queued"   -> return Queued
+      "queued"       -> return Queued
+
+      "unpinned"     -> return $ Unexpected "unpinned"
+      "unpin_queued" -> return $ Unexpected "unpin_queued"
+      "remote"       -> return $ Unexpected "remote"
+
+      _              -> return $ FailedWith errorTxt
```

Comment:
The cluster daemon shoves everything into one flat type (except when it doesn't, see `"queued"` in their source), sooooo I guess we're handling a bunch of those cases 🤷‍♀️